### PR TITLE
Allow building with GHC 9.8

### PR DESCRIPTION
**Pull request by RyanGlScott**
For the most part, this is a matter of bumping submodules and raising upper version bounds.

### CODE REVIEW

**Review by mccleeary-galois — APPROVED** [review]

File: .github/workflows/ci-matrix.yml [hunk 1]
```diff
@@ -65,3 +65,10 @@ jobs:
       if: runner.os == 'Linux'
       run: |
         cabal v2-test pkg:dismantle-${{ matrix.arch }}
+
+    - uses: actions/cache/save@v4
+      name: Save cabal store cache
+      if: always()
```
Comment (kquick) [inline .github/workflows/ci-matrix.yml:71]: Is there a reason for this `if` statement?

**Review by kquick — APPROVED** [review]

File: .github/workflows/ci-matrix.yml (hunk 1 above)
Comment (RyanGlScott) [reply .github/workflows/ci-matrix.yml:71]: Yes, this is needed to ensure that we always save the cache, even if a previous step in the workflow fails. Without this, the cache-saving step would be skipped on failure, thus needing to rebuild all of the dependencies on subsequent jobs until you get a successful one. (This is a rather baffling default that GitHub Actions' `cache` action uses, which I always make sure to override.)

Comment (kquick) [reply .github/workflows/ci-matrix.yml:71]: TIL.  Can you add a comment to preserve this info?

Comment (RyanGlScott) [reply .github/workflows/ci-matrix.yml:71]: Sure.

It's worth noting that I have copy-pasted this override in nearly every GitHub Actions YAML file I've ever touched in the `GaloisInc` organization, which means that we really ought to include the same comment in all of those other places... oh well. I'll make sure to add it to future YAML-related changes going forward.

Comment (kquick) [reply .github/workflows/ci-matrix.yml:71]: Yeah, I just noticed this part; it's too easy to skim over these "boilerplate" portions..  I wish there was a better way to get some commonality, but I'm not sure that the cost/benefit of trying to create our own consolidation action is positive.

**Pull request merged**